client/verta/verta/endpoint/_endpoint.py

```diff
@@ -508,6 +508,9 @@ def _set_kafka_settings(self, stage_id, kafka_settings):
                 "`kafka_settings` must be of type verta.endpoint.KafkaSettings,"
                 " not {}".format(type(kafka_settings))
             )
+        if kafka_settings._cluster_config_id is None:
+            config = kafka.list_kafka_configurations(self._conn)[0]
+            kafka_settings._cluster_config_id = config.get("id", None) if config else None
```

Comment:
What error does the backend return if we pass `None` as the config ID?

Reply:
I'm asking because I feel like if `config` doesn't contain `"id"` (`config.get("id", None)`), or if `config` is somehow empty (`if config`), tyhen that should be an error condition rather than just defaulting to `None`.

But if the backend will return an error a couple lines below, then that's probably fine.

Reply:
How do you feel about this:

I think it's a better practice for the client to explicitly raise an error here saying "We expected a non-empty config with the backend" or "The config should've contained an ID, but didn't". Rather than silently set it to `None` and rely on a subsequent call to the backend to raise a `500`.

We already know our program's state here is going to result in an issue; we shouldn't have it continue running.

And I don't think a client should rely on the backend to raise an error for an issue that itself originated from the backend.

Reply:
Ok, cool.  No objections to that and it gives us an opportunity to be a bit more specific with the error message.  I'll ping you when that update is in place shortly.

Reply:
Changes and related tests are now included